woocommerce/payment-gateway/Handlers/Capture.php

```diff
@@ -0,0 +1,424 @@
+
+			$order->add_order_note( $message );
+
+			// add the standard capture data to the order
+			$this->do_capture_success( $order, $response );
+
+			// if the original auth amount has been captured, complete payment
+			if ( $this->get_gateway()->get_order_meta( $order, 'capture_total' ) >= Framework\SV_WC_Helper::number_format( $this->get_order_authorization_amount( $order ) ) ) {
+
+				// prevent stock from being reduced when payment is completed as this is done when the charge was authorized
+				add_filter( 'woocommerce_payment_complete_reduce_order_stock', '__return_false', 100 );
+
+				// complete the order
+				$order->payment_complete();
+			}
+
+			return array(
+				'success' => true,
+				'code'    => 200,
+				'message' => $message,
+			);
+
+		} catch ( Framework\SV_WC_Plugin_Exception $exception ) {
+
+			// add an order note if this isn't a general error
+			if ( 500 !== $exception->getCode() ) {
+
+				$note_message = sprintf(
+				/* translators: Placeholders: %1$s - payment gateway title (such as Authorize.net, Braintree, etc), %2$s - failure message. Definitions: "capture" as in capturing funds from a credit card. */
+					__( '%1$s Capture Failed: %2$s', 'woocommerce-plugin-framework' ),
+					$this->get_gateway()->get_method_title(),
+					$exception->getMessage()
+				);
+
+				$order->add_order_note( $note_message );
+			}
+
+			return array(
+				'success' => false,
+				'code'    => $exception->getCode(),
+				'message' => $exception->getMessage(),
+			);
+		}
+	}
+
+
+	/**
+	 * Adds the standard capture data to an order.
+	 *
+	 * @since 5.3.0-dev
+	 *
+	 * @param \WC_Order $order the order object
+	 * @param Framework\SV_WC_Payment_Gateway_API_Response $response transaction response
+	 */
+	public function do_capture_success( \WC_Order $order, Framework\SV_WC_Payment_Gateway_API_Response $response ) {
+
+		$total_captured = (float) $this->get_gateway()->get_order_meta( $order, 'capture_total' ) + (float) $order->capture->amount;
+
+		$this->get_gateway()->update_order_meta( $order, 'capture_total',   Framework\SV_WC_Helper::number_format( $total_captured ) );
+		$this->get_gateway()->update_order_meta( $order, 'charge_captured', $this->get_gateway()->supports_credit_card_partial_capture() && $this->get_gateway()->is_partial_capture_enabled() && $total_captured < (float) $this->get_order_capture_maximum( $order ) ? 'partial' : 'yes' );
+
+		// add capture transaction ID
+		if ( $response && $response->get_transaction_id() ) {
+			$this->get_gateway()->update_order_meta( $order, 'capture_trans_id', $response->get_transaction_id() );
+		}
+	}
+
+
+	/**
+	 * Lets gateways handle any specific capture failure results for the order.
+	 *
+	 * @since 5.3.0-dev
+	 *
+	 * @param \WC_Order $order WooCommerce order object
+	 * @param Framework\SV_WC_Payment_Gateway_API_Response $response API response object
+	 */
+	public function do_capture_failed( \WC_Order $order, Framework\SV_WC_Payment_Gateway_API_Response $response ) { }
```

Comment:
I suppose this can't/shouldn't be an abstract child classes must implement?

Reply:
nah, most plugins won't have any use for it but it still needs to exist just in case the plugin needs to do any special handling based on the API response code or something